examples/deployment_examples.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import path
+
+import yaml
+
+from kubernetes import client, config
+
+DEPLOYMENT_NAME = "nginx-deployment"
+
+
+def create_deployment_object():
+    # Instantiate an empty deployment object
+    deployment = client.ExtensionsV1beta1Deployment()
+    # Fill required Deployment fields (apiVersion, kind and metadata)
+    deployment.api_version = "extensions/v1beta1"
+    deployment.kind = "Deployment"
+    deployment.metadata = client.V1ObjectMeta(name=DEPLOYMENT_NAME)
+    # Create and configurate a spec section
+    spec = client.ExtensionsV1beta1DeploymentSpec()
+    spec.replicas = 3
+    spec.template = client.V1PodTemplateSpec()
+    spec.template.metadata = client.V1ObjectMeta(labels={"app": "nginx"})
+    spec.template.spec = client.V1PodSpec()
+    # Configureate Pod template container
+    container = client.V1Container()
+    container.name = "nginx"
+    container.image = "nginx:1.7.9"
+    contianer.ports = [client.V1containerPort(container_port=80)]
+    spec.template.spec.containers = [container]
+    # Assign spec section into deployment.spec
+    deployment.spec = spec
+
+    return deployment
+
+
+def create_deployment(api_instance, deployment):
+    # Create deployement
+    api_response = api_instance.create_namespaced_deployment(
+        body=deployment,
+        namespace="default")
+    print("Deployment created. status='%s'" % str(api_response.status))
+
+
+def update_deployment(api_instance, deployment):
+    # Update container image
+    deployment.container.image = "nginx:1.9.1"
+    # Update the deployment
+    api_response = api_instance.replace_namespaced_deployment(
```

Comment:
I suggest you use `patch_namespaced_deployment` here as it is the recommended way to update an API object. Replace can have unintentional consequences.

Reply:
Good suggestion.